sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/models/HttpRequestContext.java

```diff
@@ -32,8 +32,9 @@ public final class HttpRequestContext {
     // It will likely have one or more substitutions in it, which will be replaced with the appropriate parameter values
     // annotated with @PathParam.
     private String path;
+    private String[] headers;
 
-    private final Map<String, String> headers;
+    private final Map<String, String> headerParams;
```

Comment:
Why do we need 2 collections of headers? We have `String[]` and `Map<String, String>`. This is a single request context, so, we should be able to merge the static and dynamic header params into one collection.